flocker/control/_config.py

```diff
@@ -1283,7 +1295,7 @@ def model_from_configuration(applications, deployment_configuration):
     :returns: A ``Deployment`` object.
     """
     nodes = deployment_from_configuration(
-        deployment_configuration, applications)
+        deployment_state, deployment_configuration, applications)
     return Deployment(nodes=frozenset(nodes))
```

Comment:
`model_from_configuration` seems a bit pointless now.
Why can't we ditch it and have `deployment_from_configuration` return a `Deployment`? ....which would make more sense, given its name.